### PR TITLE
fix: error propogation and update compelete events

### PR DESCRIPTION
**Pull request by Ash-exp**
This PR fixes the issue: if any task gets fail in pre/post CI/CD we are considering that we have no artifcats to download

Fixes https://github.com/devtron-labs/devtron/issues/5535

### CODE REVIEW

File: util/Util.go [hunk 1]
```diff
@@ -249,3 +250,16 @@ func CreateAndWriteFile(filePath string, content string) error {
...
+func PopulateStageError(err error) {
+	if err == nil {
+		return
+	}
+	if _, fileErr := os.Stat(TerminalLogDir); os.IsNotExist(fileErr) {
+		_ = os.Mkdir(TerminalLogDir, os.ModeDir)
+	}
+	writeErr := os.WriteFile(path.Join(TerminalLogDir, TerminalLogFile), []byte(err.Error()), os.ModePerm)
+	if writeErr != nil {
+		log.Println(util.DEVTRON, writeErr)
```
Comment (gireesh-naidu) [inline util/Util.go:263]: log some contextual message here

File: util/CommonConstants.go [hunk 2]
```diff
@@ -44,6 +46,9 @@ const (
 	TeeCommand                   = "tee"
 	LogFileName                  = "main.log"
 	NewLineChar                  = "\n"
+	ArtifactSourceType           = "CI-RUNNER"
+	ArtifactMaterialType         = "git"
+	TerminationLogDir            = "/dev/termination-log"
```
Comment (gireesh-naidu) [inline util/CommonConstants.go:51]: I think this is the default log location path, we can move this constant to common-lib and add the same path in ci-runner pod template from devtron swell

File: util/CommonConstants.go [hunk 3]
```diff
@@ -26,6 +26,8 @@ const (
 	DOCKER_PS_START_WAIT_SECONDS = 150
 	HOMEDIR                      = "/"
 	WORKINGDIR                   = "/devtroncd"
+	TerminalLogDir               = "/dev"
+	TerminalLogFile              = "/termination-log"
```
Comment (gireesh-naidu) [inline util/CommonConstants.go:30]: we should construct the below ``TerminationLogDir`` using these vars

File: helper/EventHelper.go [hunk 4]
```diff
@@ -720,7 +780,7 @@ func GetImageScanningEvent(ciCdRequest CommonWorkflowRequest) ImageScanningEvent
 		Image:        ciCdRequest.CiArtifactDTO.Image,
 		Digest:       ciCdRequest.CiArtifactDTO.ImageDigest,
 	}
-	var stage NotifyPipelineType
+	var stage PipelineType
```
Comment (gireesh-naidu) [inline helper/EventHelper.go:783]: this can be private

**Review by gireesh-naidu — DISMISSED** [review]

**Review by vikramdevtron — APPROVED** [review]

**Review by vikramdevtron — APPROVED** [review]

**Pull request merged**